### DISCUSSION TITLE
Adding Component In Placeholder Not Working in Experience Editor

### DISCUSSION THREAD

**Discussion by parvathisomanahallicnx** [Sitecore/jss#1921] [Help]
### Description

I am trying to add a component under placeholder in experience editor, I am getting error " Could not find the rendering in the HTML loaded from server" , attached the screens.

"renderingDomElement" is null so getting the error.

Its Sitecore 9.3, Sitecore JSS 13.0.0, React frontend

When i debug in browser, I see that <code> element for newly adding component is not generated, i see this has been generated in other Sitecore JSS project. I assume because <code> element is not generating and so required ID value is not present in page html so it throws null for renderingDomElement.

**How to Reproduce the issue:**
Open page in experience editor, click on "add here",  selected component in "Select a Rendering" wizard, select datasource, click on okay, then the error will be displayed.

<img width="452" alt="Picture4" src="https://github.com/user-attachments/assets/b531ddfe-1c8f-4910-8128-d348df14ed6f">
<img width="452" alt="Picture5" src="https://github.com/user-attachments/assets/e3679b03-663b-417e-8b88-811c0e09b784">

**Comment by illiakovalenko** [comment]
Closed this discussion since the related issue was created
